Fix contact photos after #32057

Pytal opened the pull request:
Followup to #32057

| | UID | X-NEXTCLOUD-UID
--- | --- | ---
**Before** #32057 | John | Not set
**After** #32057 | bf2b3dc13e308eafc1bb824842d3ab91 | John

Before | After
--- | ---
![image](https://user-images.githubusercontent.com/24800714/211951088-db402ca8-0e5f-4bec-8349-0732dc6011f2.png) | ![image](https://user-images.githubusercontent.com/24800714/211951096-9039783a-e905-4592-b0f7-9d2e38c6a5fc.png)

Pytal committed:
Fix contact photos

Signed-off-by: Christopher Ng <chrng8@gmail.com>

szaimen commented:
/backport to stable25

nickvergessen commented on lib/private/Contacts/ContactsMenu/ContactsStore.php:
```diff
@@ -284,8 +284,8 @@ public function findOne(IUser $user, int $shareType, string $shareWith): ?IEntry
 
-		if (isset($contact['UID'])) {
-			$uid = $contact['UID'];
+		$uid = $contact['X-NEXTCLOUD-UID'] ?? $contact['UID'];
```
Undefined index `UID`

But there are more errors caused by https://github.com/nextcloud/server/pull/32057
WIll push to your PR in a moment

nickvergessen committed:
Map the X-NEXTCLOUD-UID contact property to the UID property of IEntry

This restores the previous behaviour from before
https://github.com/nextcloud/server/pull/32057

Signed-off-by: Joas Schilling <coding@schilljs.com>

nickvergessen commented:
In my commit I also remapped the UID of IEntry to return the user id again, to avoid breaking change of the API and all IProviders.
https://github.com/nextcloud/server/pull/32057#issuecomment-1380928134

nickvergessen commented:
Avatar menus still don't work. They return 404 on the `POST findOne` request.
The addressbook search used by the ContactsMenu\Manager does not return an entry anymore.
At this point I wonder if it would be better to revert https://github.com/nextcloud/server/pull/32057 and have a look for another or complete fix with @miaulalala

Pytal commented:
> Avatar menus still don't work. They return 404 on the `POST findOne` request. The addressbook search used by the ContactsMenu\Manager does not return an entry anymore. At this point I wonder if it would be better to revert #32057 and have a look for another or complete fix with @miaulalala

Contact actions are also broken, reverting https://github.com/nextcloud/server/pull/32057 first sounds good then
